docs/get-details/ethereum_to_algorand.md

```diff
@@ -0,0 +1,231 @@
+
+## Availability of Resources to Smart Contract
+
+Algorand is designed to ensure high TPS and low latency. Every access to the blockchain state is very costly (in time). Therefore, to ensure blockchain state access is not in the critical path of block evaluation and to limit its cost, Algorand applications are restricted in the amount of resources (account balance/state, application state, ...) they can access. Furthermore, these resources need to be specified inside the transaction. This is to allow nodes to pre-fetch these data.
+
+While this may seem a strong restriction, most smart contracts do not dynamically decide which resources to access depending on the current blockchain state. In other words, it is possible to know in advance which resources will be accessed.
+
+In the near future, the [simulate endpoint](https://medium.com/algorand/try-before-you-buy-on-algorand-5acd1b9617d1) will allow a dApp to easily and dynamically learns the list of resources that need to be added to a transaction.
+
+## Smart Contract Storage
+
+One important difference between Ethereum and Algorand smart contract is storage.
+
+Ethereum smart contract storage is a huge array of 2^256 uint256 elements. The solidity language has higher-level types like dynamic arrays and mappings that are then mapped to this huge storage array (dynamic types use keccak to compute the location of each item).
+
+For performances reasons, Algorand smart contract have [three different types of storage](../dapps/smart-contracts/apps/state). While it is possible to only use boxes and essentially have a similar model as Ethereum (with the caveat that the boxes used need to be specified in the transaction), it can be more cost-effective to use local and global storage in some cases.
+
+In particular, the common solidity pattern
+
+```solidity
+mapping (address => uint) public balances;
+```
+
+is often better replaced by local storage. (With the caveat that local storage can always be erased by the account holder using [ClearState transactions](../dapps/smart-contracts/apps/). So care should be taken that doing so would not put the smart contract in an insecure state.)
+
+## Unique Features of Algorand: Multisig, Atomic Transfers, Rekeying
+
+### Multisig Accounts
+
+On Ethereum, it is possible to write smart contracts to ensure that fund transfers requires approval/signature by multiple distinct users. On Algorand, multisig accounts are first-class citizens are can be created very easily. See [the multisig account documentation](../accounts/create/#multisignature).
+
+### Atomic Transfer / Group Transaction
+
+Atomic transfers or group transactions allow to group transacitons together so that either all succeed or all fail. This can allow two users to securely exchange assets without risking one user to receive the asset without ever sending back its own assets.
```

Comment:
```suggestion
Atomic transfers or group transactions allow the grouping of multiple transactions together so that they either all succeed or all fail. This can allow two users to securely exchange assets without the risk of one of the users failing to fulfil side of the transaction.
```

Reply:
Looks good to me.